src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl extends AbstractDao implements MovieSessionDao {
+    public MovieSessionDaoImpl() {
+        super(HibernateUtil.getSessionFactory());
+    }
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return super.create(movieSession);
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        return super.get(MovieSession.class, id);
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            LocalDateTime startOfDay = date.atStartOfDay();
+            LocalDateTime endOfDay = date.atTime(LocalTime.MAX);
+            Query<MovieSession> query = session.createQuery("from MovieSession "
+                            + "where movie.id = :movieId and "
+                            + "showTime between :startOfDay and :endOfDay "
+                            + "order by showTime",
+                    MovieSession.class);
+            query.setParameter("movieId", movieId);
+            query.setParameter("startOfDay", startOfDay);
+            query.setParameter("endOfDay", endOfDay);
+            return query.getResultList();
```

Comment:
Don't get all `MovieSession` from dao and then sort them on service layer in `findAvailableSessions()`. It's better for performance to extract "ready to use" data from DB on dao layer. In this case, you are correctly extracting and sorting data in the DAO layer itself, which is good for performance. Ensure that this approach is maintained and not altered in the service layer.